src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,208 @@
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  _expand (heading: HTMLElement) {
+
+    if (heading === undefined) {
+      return;
+    }
+
+    const content = heading.nextElementSibling;
+
+    // heading elment should always have nextElementSibling (checked on _childrenChange)
+    // but in case it is null, return.
+    if (content === null) {
+      return;
+    }
+
+    // toggle expanded and aria-expanded attoributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded');
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '');
+      content.setAttribute('aria-expanded', 'true');
+    }
+  }
+
+  // children of multi-panel should always be even nuber (heading/content pair)
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error(`detected odd number of elements inside multi-panel,
+      please make sure you have heading/content pair`);
+    }
+
+    const children : Element[] = Array.from(this.children);
+    let tabindexAlreadySet : boolean = false;
+
+    for (let i = 0; i < children.length; i += 2) {
+      const heading = children[i];
+      const content = children[i + 1];
+      const randomId = Math.random().toString(36).substr(2, 9);
+
+      // if panel-heading/panel-content class is already assigned to elements
+      // (= new child was added earlier), skip the ID/tabindex setting activities
+      // and just flag tabindexAlreadySet
+      if (heading.classList.contains('panel-heading')
+        && content.classList.contains('panel-content')) {
+        tabindexAlreadySet = true;
+        continue;
+      }
+
+      // for A11y add id and tab index
+      heading.classList.add('panel-heading');
+      heading.setAttribute('tabindex', '-1');
+      heading.id = `panel-heading-${randomId}`;
+      heading.setAttribute('aria-controls', `panel-content-${randomId}`);
+
+      content.classList.add('panel-content');
+      content.id = `panel-content-${randomId}`;
+      content.setAttribute('aria-labelledby', `panel-heading-${randomId}`);
+    }
+
+    // Only if tab index was never set (= initial load) make the first heading focusable.
+    // otherwise keep tab index where already is assigned.
+    if (!tabindexAlreadySet) {
+      children[0].setAttribute('tabindex', '0');
+    }
+  }
+
+  _getClosestHeading (el: Element) {
+    const closestEl = el.closest('multi-panel > *');
+    if (closestEl && closestEl.classList.contains('panel-heading')) {
+      return closestEl as HTMLElement;
+    }
+    return undefined;
+  }
```

Comment:
This function doesn't use any state from the class (no `this.`), so it could just be a function.

Reply:
.... I get your point but then `_prevHeading()` `_firstHeading()` `_lastHeading()` are all in there as method because they use `this`.... and I kind of want to keep them all in one place... is that bad...